<?xml version="1.0"?>
<launch>
  <arg name="arm_controller_type" default="position" />
  <param name="arm_controller_type" value="$(arg arm_controller_type)" />
  <group ns="pose_tracking_node">
    <rosparam command="load" file="$(find stir_ros)/config/moveit_servo.yaml" />
    <rosparam command="load" file="$(find stir_ros)/config/pose_tracking_settings.yaml" />
    <group if="$(eval arg('arm_controller_type') == 'position')">
      <param name="controller_command_topic" value="/crane_x7/arm_position_controller/command" />
      <param name="command_out_topic" value="/moveit_servo_command" />
      <param name="command_out_type" value="std_msgs/Float64MultiArray" />
      <node name="command_order_sorter" pkg="stir_ros" type="command_order_sorter.py" output="screen" />
    </group>
    <group if="$(eval arg('arm_controller_type') == 'trajectory')">
      <param name="command_out_topic" value="/crane_x7/arm_position_controller/command" />
      <param name="command_out_type" value="trajectory_msgs/JointTrajectory" />
    </group>
  </group>
  <node name="pose_tracking_node" pkg="stir_ros" type="pose_tracking_node" output="screen" />
</launch>
